### PR TITLE
RavenDB-6589 Raft: Need to handle the case where a node was removed f…

### PR DESCRIPTION
**Pull request by karmeli87**
…rom raft cluster

- Even if node distribution is off, we move bad nodes to rehab and try to recover them.
- Allow refresh the database record features for the last index. Needed if we transit to/from passive.
- Dispose replication connection to to all nodes if we are in passive state.

### CODE REVIEW

File: src/Raven.Server/Documents/DocumentDatabase.cs [hunk 1]
```diff
@@ -715,7 +715,8 @@ private void NotifyFeaturesAboutStateChange(DatabaseRecord record, long index)
                 Debug.Assert(string.Equals(Name, record.DatabaseName, StringComparison.OrdinalIgnoreCase),
                     $"{Name} != {record.DatabaseName}");
 
-                if (LastDatabaseRecordIndex >= index)
+                // index and LastDatabaseRecordIndex could have equal values when we transit from/to passive and want to update the tasks. 
+                if (LastDatabaseRecordIndex > index)
```
Comment (ayende) [inline src/Raven.Server/Documents/DocumentDatabase.cs:719]: Are we handling the case of a node going away here and not replicating _to_ it?

File: src/Raven.Server/Rachis/RachisConsensus.cs [hunk 2]
```diff
@@ -475,7 +477,17 @@ public class StateTransition
             {
                 if (tx is LowLevelTransaction llt && llt.Committed)
                 {
-                    beforeStateChangedEvent?.Invoke();
+                    try
+                    {
+                        beforeStateChangedEvent?.Invoke();
+                    }
+                    catch (Exception e)
```
Comment (ayende) [inline src/Raven.Server/Rachis/RachisConsensus.cs:484]: What can fail here?

File: src/Raven.Server/Documents/DocumentDatabase.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/DocumentDatabase.cs:719]: Updating the cluster topology increase the raft index, so we had no problem in the case of not replicating *to*. The problem was to handle this on the passive node.

Comment (ayende) [reply src/Raven.Server/Documents/DocumentDatabase.cs:719]: What will cause us to _stop_ replicating to a  removed node?
The node is still in the topology, and I don't think the Loader will have any reason to remove it.

Comment (karmeli87) [reply src/Raven.Server/Documents/DocumentDatabase.cs:719]: We notice it is not in the cluster

Comment (ayende) [reply src/Raven.Server/Documents/DocumentDatabase.cs:719]: where is that done?

Comment (karmeli87) [reply src/Raven.Server/Documents/DocumentDatabase.cs:719]: https://github.com/ravendb/ravendb/blob/888a9538d659374fc585226fce1248492b7f4e5f/src/Raven.Client/ServerWide/DatabaseTopology.cs#L111

**Pull request merged**